src/NHibernate/Collection/Generic/PersistentGenericMap.cs

```diff
@@ -568,5 +578,95 @@ public void Operate()
 		}
 
 		#endregion
+
+		[Serializable]
+		private class ValuesWrapper : ICollection<TValue>, IQueryable<TValue>
+		{
+			private readonly PersistentGenericMap<TKey, TValue> _map;
+
+			public ValuesWrapper(PersistentGenericMap<TKey, TValue> map)
+			{
+				_map = map;
```

Comment:
Just storing the persistent map, not the values, since the underlying map may still be changed at that point.